om/hash.go

```diff
@@ -146,6 +148,77 @@ func (r *HashRepository[T]) CreateIndex(ctx context.Context, cmdFn func(schema F
 	return r.client.Do(ctx, cmdFn(r.client.B().FtCreate().Index(r.idx).OnHash().Prefix(1).Prefix(r.prefix+":").Schema())).Error()
 }
 
+// CreateAndAliasIndex creates a new index, aliases it, and drops the old index if needed.
+func (r *HashRepository[T]) CreateAndAliasIndex(ctx context.Context, cmdFn func(schema FtCreateSchema) rueidis.Completed) error {
+	alias := r.idx
+
+	var currentIndex string
+	aliasExists := false
+	infoCmd := r.client.B().FtInfo().Index(alias).Build()
+	infoResp, err := r.client.Do(ctx, infoCmd).ToMap()
+	if err != nil {
+		if strings.Contains(err.Error(), "Unknown index name") {
```

Comment:
I think this should work @rueian

Reply:
ok. please write tests for that.

Reply:
Hey @rueian Sorry been busy with work, Will do this tomorrow